### PR TITLE
add hilt

### PR DESCRIPTION
**Pull request by fatimaNoori77**
#5 issue finished 🥳

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -14,7 +16,6 @@ android {
         versionCode 1
         versionName "1.0"
 
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (Kaaveh) [inline app/build.gradle:17]: Why did you delete this line?

File: app/src/main/java/ir/noori/learningandroid/user/Repository.kt [hunk 2]
```diff
@@ -1,7 +1,8 @@
 package ir.noori.learningandroid.user
 
 import ir.noori.learningandroid.RetrofitInstance
+import javax.inject.Inject
 
-class Repository {
+class Repository @Inject constructor() {
     fun fetchUsers()= RetrofitInstance.retrofit.fetchUsers()
```
Comment (Kaaveh) [inline app/src/main/java/ir/noori/learningandroid/user/Repository.kt]: Provide retrofit instance with hilt.

**Review by Kaaveh — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (fatimaNoori77) [reply app/build.gradle:17]: I had an error that I thought was due to this. I deleted it, but I forgot to put it back. It will be fixed in the next commit

File: app/src/main/java/ir/noori/learningandroid/user/Repository.kt (hunk 2 above)
Comment (fatimaNoori77) [reply app/src/main/java/ir/noori/learningandroid/user/Repository.kt]: done

File: app/src/main/java/ir/noori/learningandroid/RetrofitInstance.kt [hunk 3]
```diff
@@ -1,26 +1,41 @@
...
 import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Singleton
 
+@Module
+@InstallIn(SingletonComponent::class)
 object RetrofitInstance {
 
-    const val BASE_URL = "https://jsonplaceholder.typicode.com/"
+    @Provides
+    fun provideBaseUrl() = "https://jsonplaceholder.typicode.com/"
```
Comment (Kaaveh) [inline app/src/main/java/ir/noori/learningandroid/RetrofitInstance.kt]: You don't need to provide the base URL. Just define it as a `const` like before

File: app/src/main/java/ir/noori/learningandroid/user/ui/UsersAdapter.kt [hunk 4]
```diff
@@ -4,10 +4,11 @@ import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 import ir.noori.learningandroid.MyApplication
-import ir.noori.learningandroid.user.data.entity.UsersModel
 import ir.noori.learningandroid.databinding.ItemUserBinding
+import ir.noori.learningandroid.user.data.entity.UsersModel
+import javax.inject.Inject
 
-class UsersAdapter(val list : ArrayList<UsersModel>) :
+class UsersAdapter @Inject constructor(val list : ArrayList<UsersModel>) :
```
Comment (Kaaveh) [inline app/src/main/java/ir/noori/learningandroid/user/ui/UsersAdapter.kt]: There is no need for the `@Inject` for your adapter.

**Review by Kaaveh — CHANGES_REQUESTED** [review]

File: app/src/main/java/ir/noori/learningandroid/user/ui/UsersAdapter.kt (hunk 4 above)
Comment (fatimaNoori77) [reply app/src/main/java/ir/noori/learningandroid/user/ui/UsersAdapter.kt]: done

File: app/src/main/java/ir/noori/learningandroid/RetrofitInstance.kt (hunk 3 above)
Comment (fatimaNoori77) [reply app/src/main/java/ir/noori/learningandroid/RetrofitInstance.kt]: rolled back...

**Review by Kaaveh — APPROVED** [review]
